build.py

```diff
@@ -38,3 +42,71 @@ def initialize(project):
 
     project.get_property('filter_resources_glob').extend(
         ['**/c_bastion/__init__.py'])
+
+
+@init(environments='travis')
+def set_properties_for_travis_builds(project):
+    project.version = '%{0}.%{1}'.format(
+        project.version, os.environ.get('TRAVIS_BUILD_NUMBER', 0))
+
+
+@task
+def project_version(project, logger):
+    print(project.version)
+
+
+def docker_execute(command_list, logger):
+    """ Run and tail a docker command. """
+    import sh
+    try:
+        running_command = sh.docker(command_list, _iter=True)
+        for line in running_command:
+            logger.info(line.strip())
+    except KeyboardInterrupt:
+        logger.info("Requested to terminate running docker command.")
+        running_command.process.terminate()
+
+
+@task
+def docker_run(project, logger):
+    logger.info("Running the docker image.")
+    AUTH_URL = os.environ['AUTH_URL']
+    docker_execute(['run',
+                    '--rm',
+                    '-p', '8080:8080',
+                    '-p', '2222:2222',
+                    '-e', 'AUTH_URL={0}'.format(AUTH_URL),
+                    '--name', docker_name,
+                    'c-bastion:latest',
+                    ], logger)
+
+
+@task
+@depends("clean", "package")
+def docker_build(project, logger):
+    logger.info("Building the docker image.")
+    docker_execute(['build',
+                    '--force-rm',
+                    '-t',
+                    'c-bastion:{0}'.format(project.version),
+                    '.'], logger)
+
+
+@task
+def docker_tag_latest(project, logger):
+    logger.info("Tagging version '{0}' with 'latest'.".format(project.version))
+    docker_execute(['tag',
+                    'c-bastion:{0}'.format(project.version),
+                    'c-bastion:latest'], logger)
+
+
+@task
+@depends('run_cram_tests', 'docker_tag_latest')
+def system_tests():
```

Comment:
Why first tag latest and then run system tests? I would expect it the other way around.